source/use-cases/sensitive-data-encryption.txt

```diff
@@ -0,0 +1,312 @@
+* ssn is deterministically encrypted because deterministic encryption provides filtering ability 
+* patientInfo.phone is deterministically encrypted for the same reason as ssn, and also it provides an example of encryption with a nested document
+* medicalRecords are randomly encrypted because entire array's are only encryptable with random encryption
+
+
+.. tabs::
+
+   tabs:
+
+     - id: java-jsonschema-generation
+       name: "Java"
+       content: |
+
+         The following code creates a collection using the schema above and inserts it into a database using MongoDB.
+         For the "Define a JSON Schema" section we've abstracted the createSecureClient, to focus on creation of a schema.
+         But that function uses getJSONSchema and passes it a dataKey id in order to create a client which uses the JSONSchema
+         we want to use. 
+
+         .. code-block:: java
+
+            import com.mongodb.*;
+            import com.mongodb.client.MongoClient;
+            import com.mongodb.client.MongoClients;
+            import com.mongodb.client.MongoCollection;
+            import com.mongodb.client.model.vault.DataKeyOptions;
+            import com.mongodb.client.vault.ClientEncryptions;
+
+            import org.bson.BsonDocument;
+
+            import java.util.Base64;
+            import java.util.Map;
+
+            import org.json.*;
+
+            public class FieldLevelEncryptionDemo {
+                static String dbName = "fieldLevelEncryptionDemo001";
+                static String collectionName = "patients";
+                public static void main(String[] args) {
+                    MongoCollection secureCollection = createSecureClient().getDatabase(dbName).getCollection(collectionName);
+                    ...
+                }
+                public static JSONObject getEncryptedField(String keyId, String chosenBsonType, Boolean isDeterministic){
+                    JSONObject binaryForKey = new JSONObject();
+                    binaryForKey.put("base64", keyId);
+                    binaryForKey.put("subType", "04");
+                    JSONObject binWrapper = new JSONObject();
+                    binWrapper.put("$binary", binaryForKey);
+                    
+                    JSONObject encryptedFieldWrap = new JSONObject();
+                    JSONObject encrypt = new JSONObject();
+
+                    encrypt.put("bsonType", chosenBsonType);
+                    if(isDeterministic){
+                        encrypt.put("algorithm", "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic");
+                    }else{
+                        encrypt.put("algorithm", "AEAD_AES_256_CBC_HMAC_SHA_512-Random");
+                    }
+                    encrypt.put("keyId", new JSONArray().put(binWrapper));
+                    encryptedFieldWrap.put("encrypt", encrypt);
+                    return encryptedFieldWrap;
```

Comment:
I don't necessarily think it would be clearer, and I also think it would be an extra & tedious step of switching over to JSONObjectBuilder. I spent some time considering the various options for generating JSON with java, and there aren't any great options, so if I had to make an executive decision on this - I say we keep it as is

Reply:
Ok, to me the builder pattern (https://en.wikipedia.org/wiki/Builder_pattern) is cleaner because you can define everything in a chain which resembles the structure of a hierarchical document. 

A builder pattern is also used for the client settings in the Java documentation of CSFLE (http://mongodb.github.io/mongo-java-driver/3.11/driver/tutorials/client-side-encryption/) and will likely be used in the section that precedes this one to prevent the code from being tedious. I am happy to help with rewriting this block if you need.

I do not mind releasing as-is, but would consider changing if we want to brush up the code to be easier to read. If you're not certain about the benefits, could you find someone on the Java driver team and see what they think?

Reply:
Sure, let's keep it as is for the PR while we're awaiting input on the java drivers team, and then if they wholeheartedly agree to use the builder pattern we can draw up another ticket to switch to that pattern.